core/db/migrate/20150309161154_ensure_payments_have_numbers.rb

```diff
@@ -1,8 +1,10 @@
 class EnsurePaymentsHaveNumbers < ActiveRecord::Migration
   def change
+    add_index :spree_payments, :number unless index_exists?(:spree_payments, :number)
     Spree::Payment.where(number: nil).find_each do |payment|
       payment.generate_number
       payment.update_columns(number: payment.number)
     end
+    remove_index :spree_payments, :number if index_exists?(:spree_payments, :number)
```

Comment:
Oh sure, I'm happy to leave that in! Wasn't sure if you wanted it there or not, but makes sense. I'll revise the PR accordingly.